pkg/roles/utils.go

```diff
@@ -58,3 +58,29 @@ func GetHcpSharedVpcPolicyDetails(r *rosa.Runtime, roleArn string) (bool, string
 
 	return existsQuery != nil, createPolicy, policyName, nil
 }
+
+func CheckIfRolesAreHcpSharedVpc(r *rosa.Runtime, roles []string) bool {
+	isHcpSharedVpc := false
+	for _, roleName := range roles {
+		ptrRoleName := roleName
```

Comment:
can we not just use `&roleName`

Reply:
Using a direct pointer of a for loop object (`for _, OBJECT := range ...`) results in a failure to build in golfing, since it is a temporary object and the memory address is freed every iteration of the for loop. I think it's called `implicit memory aliasing`